stock_storage_type/models/stock_location.py

```diff
@@ -0,0 +1,661 @@
+# Copyright 2019-2021 Camptocamp SA
+# Copyright 2019-2021 Jacques-Etienne Baudoux (BCIM) <je@bcim.be>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+import logging
+
+from psycopg2 import sql
+
+from odoo import api, fields, models
+from odoo.tools import float_compare
+
+_logger = logging.getLogger(__name__)
+
+
+class StockLocation(models.Model):
+
+    _inherit = "stock.location"
+
+    computed_storage_category_id = fields.Many2one(
+        comodel_name="stock.storage.category",
+        string="Computed Storage Category",
+        compute="_compute_computed_storage_category_id",
+        store=True,
+        recursive=True,
+        help="This represents the Storage Category that will be used. It depends either "
+        "on the category set on the location or on one of its parent.",
+    )
+    computed_storage_capacity_ids = fields.One2many(
+        related="computed_storage_category_id.capacity_ids",
+    )
+    pack_putaway_strategy = fields.Selection(
+        selection=[
+            ("none", "None"),
+            ("ordered_locations", "Ordered Children Locations"),
+        ],
+        required=True,
+        default="none",
+        string="Packs Put-Away Strategy",
+        help="This defines the storage strategy to use when packs are put "
+        "away in this location.\n"
+        "None: when a pack is moved to this location, it will not be put"
+        " away any further.\n"
+        "Ordered Children Locations: when a pack is moved to this "
+        "location, a suitable location will be searched in its children "
+        "locations according to the restrictions defined on their "
+        "respective location storage types.",
```

Comment:
As you added the package type on the product itself also, I would remove the restriction on "Packs"
```suggestion
        string="Put-Away Strategy",
        help="This defines the storage strategy based on package type to use when a product or package is put "
        "away in this location.\n"
        "None: when moved to this location, it will not be put"
        " away any further.\n"
        "Ordered Children Locations: when moved to this "
        "location, a suitable location will be searched in its children "
        "locations according to the restrictions defined on their "
        "respective location storage types.",
```